src/pveclib/vec_f128_ppc.h

```diff
@@ -4957,7 +6260,287 @@ static inline vec_xscvudqp (vui64_t int64)
       result = vec_xsiexpqp (q_sig, q_exp);
     }
 #else
-  result = f64[VEC_DW_H];
+  result = int64[VEC_DW_H];
+#endif
+  return result;
+}
+
+/** \brief VXS Scalar Convert Signed-Quadword to Quad-Precision format.
+ *
+ *  The signed quadword element of vector int128 is converted
+ *  to quad-precision format.
+ *  If the conversion is not exact the default rounding mode is
+ *  "round to Nearest, ties to even".
+ *
+ *  For POWER10 use the xscvuqqp instruction.
+ *  POWER9 only supports doubleword converts so use a combination of
+ *  two xscvudqp and xsmaddqp instructions.
+ *  For POWER8 and earlier use vector instruction generated by PVECLIB
+ *  operations.
+ *
+ *  \note The POWER8 implementation ignores the hardware rounding mode
+ *  <B>FPSCR<sub>RN</sub></B>.
+ *
+ *  \note At this point we are not trying to comply with PowerISA by
+ *  setting any FPSCR bits associated with Quad-Precision convert.
+ *  If such is required, FPFR, FR and FI can be set
+ *  using the Move To FPSCR Bit 0/1 (mtfsb[0|1]) instruction.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |   ?   | 2/cycle  |
+ *  |power9   | 44-53 |1/13cycles|
+ *
+ *  @param int128 a vector signed __int128 which is converted to QP format.
+ *  @return a __binary128 value.
+ */
+__binary128
+static inline vec_xscvsqqp (vi128_t int128)
+{
+  __binary128 result;
+#if defined (_ARCH_PWR10)  && (__GNUC__ >= 10)
+  __asm__(
+      "xscvsqqp %0,%1"
+      : "=v" (result)
+      : "v" (int128)
+      : );
+#elif defined (_ARCH_PWR9) && defined (__FLOAT128__) && (__GNUC__ > 7)
+  __binary128 hi64, lo64, i_sign;
+  __binary128 two64 = 0x1.0p64;
+  vui128_t q_sig;
+  vui32_t q_sign;
+  vui128_t q_neg;
+  vb128_t b_sign;
+  const vui32_t signmask = CONST_VINT128_W(0x80000000, 0, 0, 0);
+  // Collect the sign bit of the input value.
+  q_sign = vec_and ((vui32_t) int128, signmask);
+  // Convert 2s complement to unsigned magnitude form.
+  q_neg  = (vui128_t) vec_negsq (int128);
+  b_sign = vec_setb_sq (int128);
+  q_sig = vec_seluq ((vui128_t) int128, q_neg, b_sign);
+  // generate a signed 0.0 to use with vec_copysignf128
+  i_sign = vec_xfer_vui32t_2_bin128 (q_sign);
+  // Convert the unsigned int128 magnitude to __binary128
+  vui64_t int64 = (vui64_t) q_sig;
+  hi64 = int64[VEC_DW_H];
+  lo64 = int64[VEC_DW_L];
+  result = (hi64 * two64) + lo64;
+  // copy the __int128's sign into the __binary128 result
```

Comment:
Just noticed it here, but also see it elsewhere. Do you want leading caps for comments?  (nit)

Reply:
If it is that start of a sentence. Fixed this one.